core/src/main/scala/org/apache/gearpump/cluster/Application.scala

```diff
@@ -23,70 +23,63 @@ import org.apache.gearpump._
 import org.apache.gearpump.cluster.scheduler.Resource
 import org.apache.gearpump.jarstore.JarFileContainer
 
-case class AppJar(name: String, container: JarFileContainer)
-
 /**
- * Subtype this for user defined application
+ * This contains all information to run an application
+ *
+ * @param name: The name of this application
+ * @param conf: user configuration.
+ * @param appMaster: The class name of AppMaster Actor
+ *    The AppMaster must have a constructor this(appContext : AppMasterContext, app : Application)
  */
-trait Application {
-  val name : String
-  val appMaster : String
-  val conf: UserConfig
-}
+
+final case class Application(name : String, appMaster : String, conf: UserConfig)
```

Comment:
Application will not have any child class.

So, I don't think sealed trait is appliable here.